### PR TITLE
feat: codefix for `for await of`

### PR DESCRIPTION
**Pull request by Jack-Works**
Fixes #

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -39214,7 +39219,18 @@ namespace ts {
...
+                !!getAwaitedTypeOfPromise(type)
+                // for (const x of AsyncIterable<...>)
+                || (
+                    !allowAsyncIterables &&
+                    isForOfStatement(errorNode.parent) &&
+                    errorNode.parent.expression === errorNode &&
+                    getGlobalAsyncIterableType(/** reportErrors */ false) !== emptyGenericType &&
+                    isTypeAssignableTo(type, getGlobalAsyncIterableType(/** reportErrors */ false)
+                ));
+            return errorAndMaybeSuggestAwait(errorNode, suggestAwait, message, typeToString(type));
```
Comment (sandersn) [inline src/compiler/checker.ts:39233]: if this adds an error, can you add a compiler test that shows the code that is now an error and wasn't before?

Comment (Jack-Works) [reply src/compiler/checker.ts:39233]: this does not add a new error. it changes how `suggestAwait` is computed.

Comment (sandersn) [reply src/compiler/checker.ts:39233]: ah, I see. `errorAndMaybeSuggestAwait` always errors, but only suggests adding `await` if `suggestAwait` is true.

**Review by sandersn — APPROVED** [review]

**Comment by geritol** [comment]
This pr seems to have introduced a possibility for internal crash: https://github.com/microsoft/TypeScript/issues/57582

**Pull request merged**